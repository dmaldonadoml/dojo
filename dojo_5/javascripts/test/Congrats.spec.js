const Congrats = require('../congrats')
const Order = require('../order')

const efectyBuyEqualPay = require('./params/efectyBuyEqualPay')
const creditCardByEqualPayShippedByME = require('./params/creditCardByEqualPayShippedByME')
const creditCardBuyEqualPayWithCustomShipping = require('./params/creditCardBuyEqualPayWithCustomShipping')

const generateCongrat = order_data => () => {
  const order = Order(order_data)
  const result = order.congrat(Congrats)
  expect(result).toMatchSnapshot()
}

const tests = [
  {
    description: 'should get rendered congrats for Efecty',
    test: generateCongrat(efectyBuyEqualPay)
  }
  ,{
    description: 'should render congrats for orders paid by credit cards shipped customly',
    test: generateCongrat(creditCardBuyEqualPayWithCustomShipping)
  }
  ,{
    description: 'by credit card ship ME',
    test: generateCongrat(creditCardByEqualPayShippedByME)
  }
]

describe('Congrats', () => {
  describe("B = P", function() {
    tests.forEach(({description,test}) => it(description,test))
  })
})
