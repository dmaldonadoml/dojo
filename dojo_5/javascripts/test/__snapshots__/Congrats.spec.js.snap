// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Congrats B = P by credit card ship ME 1`] = `
Object {
  "heading": "¡Excelente compra!",
  "status": "success",
  "substatus": null,
  "title": "El jueves 2017-05-18T00:00:00.000-05:00. te llegará el producto",
}
`;

exports[`Congrats B = P should get rendered congrats for Efecty 1`] = `
Object {
  "heading": "¡Apúrate a pagar!",
  "status": "success",
  "substatus": null,
  "title": "Paga \${price} en \${paymentMethodName} para reservar tu compra",
}
`;

exports[`Congrats B = P should render congrats for orders paid by credit cards shipped customly 1`] = `
Object {
  "heading": "¡Tu pago está aprobado!",
  "status": "success",
  "substatus": null,
  "title": "Coordina con el vendedor el envío",
}
`;
